### ISSUE TITLE
undefined method `git_default_origin_branch?`

### ISSUE THREAD

**Issue by gqueiroz13** [Homebrew/brew#15245]
### `brew doctor` output

```shell
Error: undefined method `git_default_origin_branch?' for #<Pathname:0x00007f909c1c8a48>
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:548:in `block in check_tap_git_branch'
/usr/local/Homebrew/Library/Homebrew/tap.rb:750:in `block (2 levels) in each'
/usr/local/Homebrew/Library/Homebrew/tap.rb:749:in `each'
/usr/local/Homebrew/Library/Homebrew/tap.rb:749:in `block in each'
/usr/local/Homebrew/Library/Homebrew/tap.rb:748:in `each'
/usr/local/Homebrew/Library/Homebrew/tap.rb:748:in `each'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:547:in `map'
/usr/local/Homebrew/Library/Homebrew/diagnostic.rb:547:in `check_tap_git_branch'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:62:in `block in doctor'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:55:in `each'
/usr/local/Homebrew/Library/Homebrew/cmd/doctor.rb:55:in `doctor'
/usr/local/Homebrew/Library/Homebrew/brew.rb:94:in `<main>'
```

### Verification

- [X] My "`brew doctor` output" above says `Your system is ready to brew.` and am still able to reproduce my issue.
- [X] I ran `brew update` twice and am still able to reproduce my issue.
- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### `brew config` output

```shell
HOMEBREW_VERSION: 4.0.14-12-g775cddf
ORIGIN: https://github.com/Homebrew/brew
HEAD: 775cddf6d96bfcfe2111fd88445dc682452637d6
Last commit: 83 minutes ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 5f03267485e99f4f8cba6d3268cc09d006fb79d8
Core tap last commit: 9 weeks ago
Core tap branch: master
Core tap JSON: 17 Apr 12:37 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 14.0.3 build 1403
Git: 2.40.0 => /usr/local/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-x86_64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
```

### What were you trying to do (and why)?

Just run a daily update routine, everything were updated (brew itself and formulaes), however doctor returned an error.
Getting an error instead of `Your system is ready to brew`

### What happened (include all command output)?

Got an error

### What did you expect to happen?

`Your system is ready to brew`

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew update && brew upgrade && brew upgrade --cask && brew cleanup && brew autoremove && brew doctor
```

**Comment by hoium** [comment]
I'm running into the same issue on two devices `macOS: 13.3.1-arm64` and `macOS: 13.3.1-x86_64`

**Comment by dduugg** [comment]
Fixed in https://github.com/Homebrew/brew/pull/15247

**Comment by PawelLipski** [comment]
Also, getting

```
Error: undefined method `git_origin_branch' for #<Pathname:0x000055d28fad4a28>
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/dev-cmd/bump-formula-pr.rb:126:in `bump_formula_pr'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/brew.rb:94:in `<main>'
```

but it looks fixed on #15247 as well
